TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TransformedTargetRequestAndResponseList.java

```diff
@@ -0,0 +1,59 @@
+package org.opensearch.migrations.replay;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.opensearch.migrations.replay.datatypes.ByteBufList;
+import org.opensearch.migrations.replay.datatypes.HttpRequestTransformationStatus;
+
+import lombok.Getter;
+
+public class TransformedTargetRequestAndResponseList {
+
+    protected final ByteBufList requestPackets;
+
+    @Getter
+    private HttpRequestTransformationStatus transformationStatus;
+
+    @Getter
+    protected final List<AggregatedRawResponse> responseList;
+
+    public TransformedTargetRequestAndResponseList(
+        ByteBufList requestPackets,
+        HttpRequestTransformationStatus transformationStatus
+    ) {
+        this.requestPackets = requestPackets;
+        this.transformationStatus = transformationStatus;
+        this.responseList = new ArrayList<>();
+    }
+
+    public TransformedTargetRequestAndResponseList(
+        ByteBufList requestPackets,
+        HttpRequestTransformationStatus transformationStatus,
+        AggregatedRawResponse... aggregatedResponses) {
+        this(requestPackets, transformationStatus);
+        for (var r : aggregatedResponses) {
+            addResponse(r);
+        }
+    }
+
+    public void addResponse(AggregatedRawResponse r) {
+        responseList.add(r);
+    }
+
+    public List<AggregatedRawResponse> responses() {
+        return Collections.unmodifiableList(responseList);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("TransformedTargetRequestAndResponse{");
+        sb.append(", transformStatus=").append(getTransformationStatus());
+        sb.append(responseList.stream()
+            .map(AggregatedRawResponse::toString)
+            .collect(Collectors.joining("\n", "[", "]")));
+        return sb.toString();
```

Comment:
Did you intend to close this with `}`, also, i don't see needing the `, ` in transformStatus on line 53

Reply:
Removed the leading comma.  I left the prefix suffix as `[`/`]` since the contents are of a list.  Why were you asking about `}`?

Reply:
>Why were you asking about }?

The start of this string is `TransformedTargetRequestAndResponse{` and that `{` is never closed